app/controllers/bookmarks_controller.rb

```diff
@@ -0,0 +1,59 @@
+class BookmarksController < ApplicationController
+  def show
+    @bookmark = Bookmark.find(params[:id])
+  end
+
+  def new
+    @topic = Topic.find(params[:topic_id])
+    @bookmark = Bookmark.new
+  end
+
+  def edit
+    topic = Topic.find(params[:topic_id])
+    @bookmark = topic.bookmarks.find(params[:id])
+  end
+
+  def update
+    topic = Topic.find(params[:topic_id])
+    @bookmark = topic.bookmarks.find(params[:id])
+    if @bookmark.update(bookmark_params)
+      flash[:notice] = 'bookmark was saved successfully.'
+      redirect_to topic
+    else
+      flash.now[:alert] = @bookmark.errors.full_messages.first
+      render :edit
+    end
+  end
+
+  def create
+    @topic = Topic.find(params[:topic_id])
```

Comment:
This doesn't need to be an instance variable since you're not using it in any view. Change this to `topic` and be sure to update it anywhere else it's used inside this method.